### PR TITLE
Extract common code into helpers

### PR DESCRIPTION
**Pull request by thedrow**
I'm still not sure about the names.
I'm also probably not done.
Just want some initial feedback.

### CODE REVIEW

File: oauthlib/oauth2/rfc6749/endpoints/base.py [hunk 1]
```diff
@@ -39,6 +40,28 @@ def catch_errors(self):
...
+        """Raise on failed client authentication."""
+        if self.request_validator.client_authentication_required(request):
+            if not self.request_validator.authenticate_client(request):
+                log.debug('Client authentication failed, %r.', request)
+                raise InvalidClientError(request=request)
+        elif not self.request_validator.authenticate_client_id(request.client_id, request):
+            log.debug('Client authentication failed, %r.', request)
+            raise InvalidClientError(request=request)
+
+    def _raise_on_unspported_token(self, request):
```
Comment (JonathanHuot) [inline oauthlib/oauth2/rfc6749/endpoints/base.py]: typo `unspported`?

Comment (thedrow) [reply oauthlib/oauth2/rfc6749/endpoints/base.py]: Yup.

**Comment by JonathanHuot** [comment]
No better names to suggest :-)

Cleanup was necessary, please continue if you spot other obvious places.
Else LGTM

**Comment by skion** [comment]
LGTM

**Pull request merged**